virtwho/main.py

```diff
@@ -109,30 +109,40 @@ def reload(signal, stackframe):
         raise ReloadRequest()
     exit(1, status="virt-who cannot reload, exiting")
 
+def get_version():
+    version_file = os.path.join(os.path.dirname(os.path.abspath(__file__)),'version.py')
+    version = {}
+    with open(version_file) as fp:
+        exec(fp.read(), version)
+    return "virt-who %s" % version['__version__']
 
 def main():
-    logger = options = None
+    logger = effective_config = cli_options = None
     try:
-        logger, options = parse_options()
+        logger, effective_config, cli_options = parse_options()
         # We now have the effective_config
     except OptionError as e:
         print(str(e), file=sys.stderr)
         exit(1, status="virt-who can't be started: %s" % str(e))
 
+    if 'version' in cli_options and cli_options['version']:
+        print(get_version())
+        exit(os.EX_OK)
+
```

Comment:
What about this approach? Move this block of code to `parser.py` (line 486). And move method `get_version()` to `parser.py`. It will simplify `parse_options()` method. It will not be necessary to return `cli_options` to `main.py`. Another benefit is following: it will not print all errors and warnings, when `--version` is used. It will only print version to stdout.

Reply:
@jirihnidek thats actually how I implemented it first, but that did not seem like the right thing to do, a parser actually performing an action. 

anywho, if you are up for it, iv made the suggested changes